hedera-node/docs/design/services/smart-contract-service/contract-accounts-nonces.md

```diff
@@ -0,0 +1,64 @@
+| EOA transaction (`EthereumTransaction`)                                                 | EOA nonce is incremented                                                                                   | EOA nonce is incremented, updates are not externalized to Mirror Node through transaction records, but can be picked up by reading the value from the `EthereumTransaction` body | EOA nonce is incremented, updates are not externalized to Mirror Node through transaction records, but can be picked up by reading the value from the `EthereumTransaction` body |
+| EOA transaction (`ContractCall` or `ContractCreate`)                                    | -                                                                                                          | -                                                                                                                                                                                | -                                                                                                                                                                                |
+| Contract transaction resulting in `CREATE/CREATE2` (`EthereumTransaction`)              | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are not externalized to Mirror Node                      | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are externalized to Mirror Node                          |
+| Contract transaction resulting in `CREATE/CREATE2` (`ContractCall` or `ContractCreate`) | -                                                                                                          | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are not externalized to Mirror Node                      | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are externalized to Mirror Node                          |
+
+
+
+### Contract Nonce Externalization
+
+- We can keep a `ContractId -> nonce` map inside `TxnAwareRecordsHistorian`, it would be updated on each call to `AbstractRecordingCreateOperation` (possibly using its `createOperationExternalizer` field) even if the contract creation operation does not succeed
+- When the ledgers `commit()` are executed, and we go to `TxnAwareRecordsHistorian.saveExpirableTransactionRecords` method we can save the constructed map to the top-level `RecordStreamObject`
+- This would be done via setting it inside it's `ExpirableTxnRecord` in the corresponding `EvmFnResult` field
```

Comment:
Not at all a blocker for this PR, just a couple questions:
 - Is there already an appropriate field in `EvmFnResult` for the `ContractID -> nonce` map? 
 - Did you consider externalizing one nonce per `EvmFnResult` ("spread" across the appropriate child records), instead of an aggregate map in the top-level record?

Reply:
> Is there already an appropriate field in EvmFnResult for the ContractID -> nonce map?

Currently there isn't, it will need to be added so that when we get the updated protobuf definitions we will be able to set the needed value in `EvmFnResult.toGrpc()` method.

> Did you consider externalizing one nonce per EvmFnResult ("spread" across the appropriate child records), instead of an aggregate map in the top-level record?

Yes, this was raised in [a comment in the HIP PR](https://github.com/hashgraph/hedera-improvement-proposal/pull/729#discussion_r1186372249) as well and I've added an entry in the HIP's rejected ideas section that is what I've answered in the discussion as well.

Reply:
Oops didn't do my homework...